Framework/PythonInterface/mantid/api/src/Exports/MatrixWorkspace.cpp

```diff
@@ -277,6 +297,13 @@ void export_MatrixWorkspace() {
            "Return the Run object for this workspace (deprecated, use "
            ":class:`~mantid.api.MatrixWorkspace.getRun` "
            "instead)")
+      .def("binIndexOf", &binIndexOfDeprecated,
+           MatrixWorkspace_yIndexOfXOverloads(
```

Comment:
So this is not quite there as it actually links `binIndexOf` to the `yIndexOverloads` and so the deprecation warning doesn't get out. I think we'll need to create an overloader for the `binIndexOfDeprecated` function with `BOOST_PYTHON_FUNCTION_OVERLOADS`. I think we can have the `binIndex` version not have the tolerance argument as it didn't have it before.